src/rendezvous/channels/ecdhV1.ts

```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { SAS } from '@matrix-org/olm';
+
+import { RendezvousError } from '../error';
+import {
+    RendezvousCode,
+    RendezvousIntent,
+    RendezvousChannel,
+    RendezvousTransportDetails,
+    RendezvousTransport,
+    RendezvousFailureReason,
+} from '../index';
+import { SecureRendezvousChannelAlgorithm } from '.';
+import { encodeBase64, decodeBase64 } from '../../crypto/olmlib';
+import { getCrypto } from '../../utils';
+
+const subtleCrypto = (typeof window !== "undefined" && window.crypto) ?
+    (window.crypto.subtle || window.crypto.webkitSubtle) : null;
+
+export interface ECDHv1RendezvousCode extends RendezvousCode {
+    rendezvous: {
+        transport: RendezvousTransportDetails;
+        algorithm: SecureRendezvousChannelAlgorithm.ECDH_V1;
+        key: string;
+    };
+}
+
+// The underlying algorithm is the same as:
+// https://github.com/matrix-org/matrix-js-sdk/blob/75204d5cd04d67be100fca399f83b1a66ffb8118/src/crypto/verification/SAS.ts#L54-L68
```

Comment:
It seems like it should be calling this function, rather than copying the code, even if it is a (almost) one-liner

Reply:
Fixed. n.b. the function is now exported from `SAS.ts`.